lib/pbench/server/api/resources/models.py

```diff
@@ -0,0 +1,124 @@
+import os
+import jwt
+import datetime
+from dateutil import parser
+from sqlalchemy import Column, Integer, String, DateTime
+from pbench.server.api.resources.database import Base, blacklist
+
+
+class UserModel(Base):
+    """ User Model for storing user related details """
+
+    __tablename__ = "users"
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    username = Column(String(255), unique=True, nullable=False)
+    firstName = Column(String(255), unique=False, nullable=False)
+    lastName = Column(String(255), unique=False, nullable=False)
+    password = Column(String(255), nullable=False)
+    registered_on = Column(DateTime, nullable=False)
+    email = Column(String(255), unique=True, nullable=False)
+
+    def __init__(self, bcrypt, bcrypt_log_rounds, **kwargs):
+        super().__init__(**kwargs)
+        self.username = kwargs.get("username")
+        self.firstName = kwargs.get("firstName")
+        self.lastName = kwargs.get("lastName")
+        self.password = bcrypt.generate_password_hash(
+            kwargs.get("password"), bcrypt_log_rounds
+        ).decode()
+        self.email = kwargs.get("email")
+        self.registered_on = datetime.datetime.now()
+
+    def __str__(self):
+        return f"User, id: {self.id}, username: {self.username}"
+
+    def encode_auth_token(self, token_expire_duration, user_id, logger):
+        """
+        Generates the Auth Token
+        :return: string
+        """
+        if not token_expire_duration:
+            # Expiration is not defined, token is set to expire in a second
```

Comment:
This would happen only when we dont specify the token_expire_duration in config file, do we want to default it to something? I did this for the tests, so I can see the token expiring without having to wait long

Reply:
Well, generally we don't want to embed test code in the "product", and given there's no product use case for a token that expires so quickly, I don't think we want to leave this in.

We could pick some reasonable default, or simply require that the property be in the config file. The easiest way to do the latter would be to have it in the `pbench-serve-default.cfg` and let the site `pbench-server.cfg` override it if they want.